tests/cephadm/test_ceph_orch_for_monitoring_image_version.py

```diff
@@ -0,0 +1,52 @@
+import re
+from json import loads
+
+from cli.cephadm.cephadm import CephAdm
+from cli.exceptions import OperationFailedError
+
+
+def run(ceph_cluster, **kw):
+    """
+    Validate if the ceph orch ps command reflects the version
+    for monitoring stack images
+    """
+    config = kw["config"]
+    installer_node = ceph_cluster.get_nodes(role="installer")[0]
+    cephadm = CephAdm(installer_node)
+    monitoring_images = config.get("monitoring_images")
+
+    # Get the version using podman command
+    prometheus_img = monitoring_images["ose-prometheus"]
+    alertmanager_img = monitoring_images["ose-prometheus-alertmanager"]
+    node_exporter_img = monitoring_images["ose-prometheus-node-exporter"]
+
+    # Get the version using podman command
+    cmd = f'podman run -it --rm --user root --entrypoint /bin/bash {prometheus_img} -c "prometheus --version"'
```

Comment:
maybe we can loop these operations?
something like

base_cmd = 'podman run -it --rm --user root --entrypoint /bin/bash {} -c "{} --version"'

for img, mon_stack in ([prometheus_img, 'prometheus'],
                       [alertmanager_img, "alertmanager"],
                       [node_exporter_img, "node_exporter"]):
    cmd = base_cmd.format(img, mon_stack)
    out, _ = installer_node.exec_command(sudo=True, cmd=cmd)
    _version = re.search(r"version\s+(\d+\.\d+\.\d+)", out)
    
    
    Just a suggestion. IF this can't be done or will cause logic problem with existing approach, please ignore.

Reply:
Hey @pranavprakash20  I need to store the output of  re.search(r"version\s+(\d+.\d+.\d+)", out) in a variable which I can use later for validation. 
Maybe I can add somethingv as below-
```services = {
    "prometheus": "_prom_vers",
    "alertmanager": "_alert_vers",
    "node_exporter": "_node_vers"
}

for service, version_variable in services.items():
    image = monitoring_images.get(service)
    if image:
        cmd = f"podman run -it --rm --user root --entrypoint /bin/bash {image} -c \"{service} --version\""
        out, _ = installer_node.exec_command(sudo=True, cmd=cmd)
        _version = re.search(r"version\s+(\d+\.\d+\.\d+)", out)
        if _version:
            globals()[version_variable] = _version.group(1)
```
   Let me try and update the PR accordingly.

Reply:
Hey @pranavprakash20  Its complicating the code actually, because we have images also to iterate through, so I will prefer to leave it as it is, let me know if you agree.